metrics-core/src/main/java/software/amazon/swage/metrics/record/cloudwatch/MetricDataAggregator.java

```diff
@@ -74,24 +84,59 @@ public MetricDataAggregator(final DimensionMapper dimensionMapper)
      * @param value Recorded value for the metric event
      * @param unit Unit for interpreting the value
      */
-    public void add(
+    public void addAggregated(
             final TypedMap context,
             final Metric name,
             final double value,
             final StandardUnit unit)
     {
         //TODO: avoid doing this every time for a context - caching, or?
-        List<Dimension> dimensions = dimensionMapper.getDimensions(name, context);
+        final List<Dimension> dimensions = dimensionMapper.getDimensions(name, context);
+        final DatumKey key = new DatumKey(dimensions, name.toString(), unit);
 
-        DatumKey key = new DatumKey(name.toString(), unit, dimensions);
-        statisticsMap.merge(
-                key,
-                new StatisticSet()
+        statisticsMap.compute(key, (entryKey, entryValue) -> {
```

Comment:
I still think this is incorrect. 
`These are explicitly aggregated values, where only the total number of occurrences in a context matter.`
You are publishing a statistics set of the deltas not the actual value of the count within the context. Which means you lose visibility in to total count per context.

For example if in a context I record counts 100, 5, 7, -2 , -100. I end up with 5 samples in cloudwatch with a max of 100, a min of -100, an average of 2, and a sum of 10. 

If, in the same interval, in a second context (another request for example) I recorded counts 1, 2. This would end up posting a sample of 7 with a max of 100, a min of -100, an average of 1.86, and a sum of 13.

In this example, I would expect to see only 2 samples in cloudwatch: 10 and 3.

Reply:
That sounds like it's out of scope for this PR. I didn't change anything regarding contexts.

Reply:
Can you please tell me why this needs to happen in this PR when the behavior of your example remains unchanged?